nebullvm/installers/auto_installer.py

```diff
@@ -46,68 +39,75 @@
 }
 
 
-def check_backends(include_backends: Union[List[str], str]) -> List[str]:
-    if isinstance(include_backends, str) and include_backends == "all":
-        new_include_backends = list(INSTALLERS.keys())
+def select_frameworks_to_install(
+    include_frameworks: Union[List[str], str],
+    include_backends: Union[List[str], str],
+) -> List[str]:
+    if isinstance(include_frameworks, str) and include_frameworks == "all":
+        frameworks_list = list(INSTALLERS.keys())
+    elif isinstance(include_frameworks, list):
+        frameworks_list = []
+        for framework in include_frameworks:
+            if framework in INSTALLERS.keys():
+                frameworks_list.append(framework)
+            else:
+                logger.warning(f"Framework {framework} not supported")
+
+        if isinstance(include_backends, str) and include_backends == "all":
+            for framework in frameworks_list:
+                for backend in SUPPORTED_BACKENDS_DICT[framework]:
+                    frameworks_list.append(backend)
+        elif isinstance(include_backends, list):
+            for framework in frameworks_list:
+                for backend in include_backends:
+                    if backend in SUPPORTED_BACKENDS_DICT[framework]:
+                        frameworks_list.append(backend)
+        else:
+            raise ValueError("Invalid backends list")
```

Comment:
why is the wrong framework just giving a warning while the wrong backend is raising an error?

Reply:
Furthermore what if `include_backends` is not a list but just a single backend? With the current implementation it would crash right?

Reply:
When the select_frameworks_to_install function is called  include_backends can be either a list of compilers or the string "all", so it cannot happen that case